src/backend-manager/src/backend_manager.rs

```diff
@@ -265,192 +270,92 @@ impl BackendManager {
+                                self.stop_replay(id as usize).await?;
+                                return Ok(());
+                            }
                         }
-
-                        // SAFETY: The if above ensures safety
-                        let arg = unsafe { args.as_u64().unwrap_unchecked() };
-
-                        return self.stop_replay(arg as usize).await;
+                        // TODO: return error
+                        Ok(())
                     }
-
                     "ct/select-replay" => {
-                        if args.is_none() {
-                            // TODO: return error
-                        }
-                        // SAFETY: The if above ensures safety
-                        let args = unsafe { args.unwrap_unchecked() };
-
-                        if !args.is_u64() {
-                            // TODO: return error
+                        if let Some(Value::Number(num)) = args {
+                            if let Some(id) = num.as_u64() {
+                                self.select_replay(id as usize)?;
+                                return Ok(());
+                            }
                         }
-
-                        // SAFETY: The if above ensures safety
-                        let arg = unsafe { args.as_u64().unwrap_unchecked() };
-
-                        return self.select_replay(arg as usize);
+                        // TODO: return error
+                        Ok(())
                     }
-
                     _ => {
-                        if args.is_none() {
-                            // No request arguments. Forwarding...
-                            return self.message_selected(message).await;
-                        }
-
-                        // SAFETY: The if above ensures that args is not None
-                        let args = unsafe { args.unwrap_unchecked() };
-
-                        if !args.is_object() {
-                            // Irrelevant args. Forwarding...
-                            return self.message_selected(message).await;
-                        }
-
-                        // SAFETY: The if above ensures that args is an object
-                        let args = unsafe { args.as_object().unwrap_unchecked() };
-
-                        if !args.contains_key("replay-id") {
-                            // Not a request to specific backend. Forwarding...
-                            return self.message_selected(message).await;
-                        }
-
-                        // SAFETY: The if above ensures that replay-id exists
-                        let replay_id = unsafe { args.get("replay-id").unwrap_unchecked() };
-
-                        if !replay_id.is_u64() {
-                            // Expected integer ID. IDK what this is. Forwarding...
-                            return self.message_selected(message).await;
+                        if let Some(Value::Object(obj_args)) = args {
```

Comment:
that's much better: still maybe we can directly map it but in any case an improvement